### DISCUSSION TITLE
How to Use HTTP in tibber-pulse reader

### DISCUSSION THREAD

**Discussion by Clanchef** [micw/tibber-pulse-reader#10] [General]
Hallo,
Ich versuche mich gerade an http Schnittstelle des tibber-pulse-readers.
Den Webserver des Tibber Pulse habe ich aktiviert, 

Meine application.yaml

LOG_LEVEL: info
TIBBER_PULSE_SOURCE: http
TIBBER_PULSE_HOST: 192.168.178.37
TIBBER_PULSE_USERNAME: admin
TIBBER_PULSE_PASSWORD: ########
#TIBBER_PULSE_CRON: */15 * * * * *

Docker Konfig (docker compose)

version: "3"
services:
  tibber-pulse-reader:
    image:  ghcr.io/micw/tibber-pulse-reader:master
    container_name: tibber-pulse
    restart: always
    privileged: true
    network_mode: host
    volumes:
      - /home/martin/pulse/application.yaml:/application.yaml

Wie komme ich an die dekodierten Daten jetzt ran und wie konfiguriere ich den lokalen Zugriff z.B. in evcc bzw. homeassistant?

Das Logfile sieht so aus:

. ____ _ __ _ _
/\ / ' __ _ () __ __ _ \ \ \
( ( )__ | '_ | '| | ' / ` | \ \ \
\/ )| |)| | | | | || (| | ) ) ) )
' || .__|| ||| |_, | / / / /
=========||==============|/=////
:: Spring Boot :: (v2.5.14)

2023-05-25 12:17:55.582 INFO 1 --- [ main] de.wyraz.tibberpulse.TibberPulseReader : Starting TibberPulseReader v1.0.0-SNAPSHOT using Java 17-ea on Clanchef-ASUSZEN with PID 1 (/tibber-pulse-reader.jar started by root in /)
2023-05-25 12:17:55.584 INFO 1 --- [ main] de.wyraz.tibberpulse.TibberPulseReader : No active profile set, falling back to 1 default profile: "default"
2023-05-25 12:17:55.911 WARN 1 --- [ main] d.w.tibberpulse.sink.MeterDataHandler : No publishers are configured.
2023-05-25 12:17:56.123 INFO 1 --- [ main] de.wyraz.tibberpulse.TibberPulseReader : Started TibberPulseReader in 0.892 seconds (JVM running for 1.149)
2023-05-25 12:17:56.213 WARN 1 --- [ main] d.w.t.source.TibberPulseHttpReader : Invalid response from tibber pulse gateway endpoint: HTTP/1.1 404 Not Found

**Comment by micw** [comment]
Geh mal mit dem Browser auf den Pulse Bridge, lass Dir die aktuellen Daten vom Pulse anzeigen und schaue in den Entwicklertools, von welcher URL die abgerufen werden.

**Comment by Clanchef** [comment]
Ich habs hinbekommen: War ein Denkfehler von mir. Das Tool holt sich per http die Daten und schreibt diese in einen MQTT
Der CRON Aufruf scheint aber nicht zu funktionieren. Sobald man versucht das Timing zu ändern kommt eine Fehlermeldung 
#TIBBER_PULSE_CRON: */5 * * * * *

unexpected character found /(47)
 in 'reader', line 7, column 21:
    TIBBER_PULSE_CRON: */5 * * * * *

LOG_LEVEL: info

TIBBER_PULSE_SOURCE: http
TIBBER_PULSE_HOST: 192.168.178.37/data.json?node_id=3
TIBBER_PULSE_USERNAME: admin
TIBBER_PULSE_PASSWORD: ######
TIBBER_PULSE_CRON: */5 * * * * *

PUBLISH_MQTT_ENABLED: true
PUBLISH_MQTT_HOST: 192.168.178.2
PUBLISH_MQTT_PORT: 1883
PUBLISH_MQTT_TOPIC: "{meterId}/{nameOrObisCode}"
PUBLISH_MQTT_VALUE: "{value}"

**Comment by Clanchef** [comment]
Fehler gefunden. Das Beispiel im Projekt ist nicht ganz richtig beim CRON job.

LOG_LEVEL: info

TIBBER_PULSE_SOURCE: http
TIBBER_PULSE_HOST: 192.168.178.37/data.json?node_id=3
TIBBER_PULSE_USERNAME: admin
TIBBER_PULSE_PASSWORD: ------------------
TIBBER_PULSE_CRON: 0/5 * * * * *

PUBLISH_MQTT_ENABLED: true
PUBLISH_MQTT_HOST: 192.168.178.2
PUBLISH_MQTT_PORT: 1883
PUBLISH_MQTT_TLS: false
PUBLISH_MQTT_TOPIC: "{meterId}/{nameOrObisCode}"
PUBLISH_MQTT_VALUE: "{value}"

**Reply by micw** [reply]
Die Expression ist korrekt. Da es YAML ist, darf es allerings nicht mit * anfangen, da das ein Objekt-Anfang im Yaml ist. Die Expression muss in Anführungszeichen

**Reply by micw** [reply]
Hab den Hinweis in der Doku ergänzt
https://github.com/micw/tibber-pulse-reader/commit/4927a3cdf6262c86e1893bc26a0608f93cd80486

**Reply by micw** [reply]
Wie kommt es eigentlich, dass du node_id=3 hast? Hast du mehrere Pulse (bzw. mehrere Lora Geräte am Pulse-Gateway) am Start?

**Reply by Clanchef** [reply]
Um den Webserver der Bridge freizuschalten habe ich ein paar Anläufe gebraucht. Das hat offensichtlich zu Node 3 geführt.
Ich lass das jetzt mal so weil es sehr gut funktioniert.

**Comment by Clanchef** [comment]
Danke! Tolles Projekt. Tibber lokal auslesen war genau das was ich gesucht habe!
